### DISCUSSION TITLE
Structure and order your DI container

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#35] [General]
# Structure and order your DI container

Does your Dependency Injection container is one big pile of method calls one after the other?
Are there 50 lines of just AddScoped, AddTransient, and so on? Well, let's fix this.
We can utilize extension methods to make an order to that mess!

https://steven-giesel.com/blogPost/f557493c-0919-44fc-8073-40b4fb90d53e

**Comment by pwelter34** [comment]
You might want to consider the Injectio project, source generator that helps register decorated services in the dependency injection container.

https://github.com/loresoft/Injectio

~ Paul

**Reply by linkdotnet** [reply]
Nice project - I guess and hope the generator throws some compile time error when registration is missing.

**Comment by puschie286** [comment]
we use auto registration by interfaces and reflection - this should be enough for most cases and reduce explicit registrations to minimum
